h5c/vic/src/vic-webapp/src/app/interfaces/vch.ts

```diff
@@ -131,36 +142,71 @@ export interface VchApiNetwork {
     nameservers?: string[],
     static?: string
   },
-  container?: [
-    {
-      alias: string,
-      firewall: FirewallPolicyType,
-      nameservers: string[],
-      port_group: {
-        id: string,
-        name: string
-      },
-      gateway: {
-        routing_destinations: string[],
-        address: string
-      },
-      ip_ranges: string[]
-    }
-    ]
+  container?: ContainerNetworkApi[],
+  httpProxy?: string,
+  httpsProxy?: string
+}
+
+export interface ContainerNetworkApi {
+  alias?: string,
+  firewall?: FirewallPolicyType,
+  nameservers?: string[],
+  port_group: {
+    id?: string,
+    name: string
+  },
+  gateway?: {
+    routing_destinations?: string[],
+    address: string
+  },
+  ip_ranges?: string[]
+}
+
+export interface VchUiNetwork {
+  clientNetwork?: string;
+  clientNetworkGateway?: string;
+  clientNetworkIp?: string;
+  clientNetworkRouting?: string;
+  clientNetworkType?: string;
+  publicNetwork: string;
+  publicNetworkGateway?: string;
+  publicNetworkIp?: string;
+  publicNetworkType: string;
+  bridgeNetwork: string;
+  bridgeNetworkRange?: string;
+  managementNetwork?: string;
+  managementNetworkGateway?: string;
+  managementNetworkIp?: string;
+  managementNetworkRouting?: string;
+  managementNetworkType?: string;
+  containerNetworks?: VchContainerNetworkUi[];
+  dnsServer?: string[];
+  httpProxy?: string;
+  httpProxyPort?: string;
+  httpsProxy?: string;
+  httpsProxyPort?: string;
+}
+
+export interface VchContainerNetworkUi {
+  containerNetwork: string;
+  containerNetworkDns: string;
+  containerNetworkFirewall: FirewallPolicyType;
+  containerNetworkGateway?: string;
+  containerNetworkIpRange?: string;
+  containerNetworkLabel: string;
+  containerNetworkType: string;
 }
 
 // VCH Storage
 export interface VchApiStorage {
   image_stores: string[],
-  volume_stores?: [
-    {
-      datastore: string,
-      label: string
-    }
-    ],
+  volume_stores?: {
+    datastore: string;
+    label: string
+    }[],
```

Comment:
any reason to change this from the format it was before?

Reply:
Because the format it was before was wrong... It was introduced by me, but I didn't used that interface till this network commit when I figured out that it was wrong. 

I wanted to make it an array of this kind of objects -> {datastore: string; label: string}

So the correct way to do it is {datastore: string; label: string}[] and not [{datastore: string; label: string}]